hail/src/main/scala/is/hail/expr/types/physical/PBinary.scala

```diff
@@ -11,16 +11,12 @@ import is.hail.utils._
 
 import scala.reflect.{ClassTag, _}
 
-case object PBinaryOptional extends PBinary(false)
+case object PBinaryOptional extends PCanonicalBinary(false)
```

Comment:
these should be PCanonicalBinaryOptional and PCanonicalBinaryRequired and found in the PCAnonicalBinary class, I think

Reply:
still need to get rid of these